runtime/r_wrap.c

```diff
@@ -135,17 +135,17 @@ void processCmd(char *args[])
                 // Write to forked process
                 // For some reason nonblocking fd is not working correctly with fwrite. 
                 // Check shortest-scripts.sh with big batch size to recreate the problem
-                if ((len = write(outputFd, writebuffer, currWriteLen)) < 0) {
-					switch (len) {
+                if ((len = write(outputFd, writebuffer, currWriteLen)) > 0) {
+                    currWriteLen -= len; 
+                    memmove(writebuffer, writebuffer + len, currWriteLen);
```

Comment:
What is this change for?

Reply:
I think this was a small change for minor performance improvement. The processor usually speculates the true branch faster than the other one but it's not really doing much here.

Reply:
I would be really surprised if that was true (speculating faster due to such a code change) :)